build/config.props

```diff
@@ -33,12 +33,13 @@
     <VsTargetChannel Condition="'$(IsEscrowMode)' == 'true'">int.d$(VsTargetMajorVersion).$(MinorNuGetVersion)</VsTargetChannel>
 
     <!-- .NET Core SDK Insertion Logic -->
-    <!--We don't have .NET5 installed on CI, so we need to get version from CliVersionForBuilding and install it-->
-    <!--TODO: remove temporary workaround of using LockSDKVersion. Tracking issue: https://github.com/NuGet/Home/issues/9712 -->
+    <!-- We don't have .NET 5 SDK installed on CI, so we need to get version from CliVersionForBuilding and install it -->
+    <!-- TODO: remove temporary workaround of using LockSDKVersion. Tracking issue: https://github.com/NuGet/Home/issues/9712 -->
     <LockSDKVersion>true</LockSDKVersion>
-    <OverrideCliBranchForTesting Condition="'$(LockSDKVersion)' == 'false'"></OverrideCliBranchForTesting>
-    <OverrideCliBranchForTesting Condition="'$(LockSDKVersion)' == 'true'">master 5.0.100-preview.7.20319.6</OverrideCliBranchForTesting>
-    <CliVersionForBuilding>"master 5.0.100-preview.4.20258.7"</CliVersionForBuilding>
+    <OverrideCliBranchForTesting Condition="'$(LockSDKVersion)' == 'true'">master:5.0.100-preview.8.20417.9;3.1</OverrideCliBranchForTesting>
+    <OverrideCliVersionForBuilding Condition="'$(LockSDKVersion)' == 'true'">"master 5.0.100-preview.8.20417.9"</OverrideCliVersionForBuilding>
+    <CliVersionForBuilding Condition="'$(OverrideCliVersionForBuilding)' != ''">$(OverrideCliVersionForBuilding)</CliVersionForBuilding>
+    <CliVersionForBuilding Condition="'$(OverrideCliVersionForBuilding)' == ''">5.0;3.1</CliVersionForBuilding>
```

Comment:
NuGet 5.8 is no longer inserting into .NET Core 3.1, so shouldn't test against it, or need it for building. https://github.com/NuGet/NuGet.Client/pull/3570

Also, we can't build with the latest .NET 5 preview 8 or later, as it needs msbuild 16.8 or later, which is only available with preview VS 16.8, which is not available on our CI, so we're limited to .NET 5 SDK preview 7 at max, until VS 16.8 goes GA and is available on our CI agents 😞   But I think this is what you're getting from Heng's PR, so once hers is merged, I don't think you'll need these changes in your branch.